lib/_http_client.js

```diff
@@ -169,6 +172,14 @@ function ClientRequest(input, options, cb) {
   if (options.timeout !== undefined)
     this.timeout = getTimerDuration(options.timeout, 'timeout');
 
+  if (options.signal) {
+    validateAbortSignal(options.signal, 'options.signal');
+    const listener = (e) => this.destroy();
+    options.signal.addEventListener('abort', listener);
```

Comment:
Nit: perhaps use once here as well ({ once: true })?

Reply:
I feel like it would be more hacky since we're removing the listener below even if it is invoked since `close` should be fired.

If I pass `once` and someone `aborts`s the controller - the signal will fire `abort`, the listener will be removed, then `close` will fire and we will try removing a non-existing listener which will _technically_ be no harm but will be slightly confusing.